x-pack/plugins/security_solution/public/timelines/components/timeline/insert_timeline_popover/use_insert_timeline.tsx

```diff
@@ -29,14 +40,31 @@ export const useInsertTimeline = <T extends FormData>(form: FormHook<T>, fieldNa
           : `[${currentValue.slice(cursorPosition.start, cursorPosition.end)}](${builtLink})`,
         currentValue.slice(cursorPosition.end),
       ].join('');
+
       form.setFieldValue(fieldName, newValue);
     },
     [basePath, cursorPosition, fieldName, form]
   );
+
   const handleCursorChange = useCallback((cp: CursorPosition) => {
     setCursorPosition(cp);
   }, []);
 
+  // insertTimeline selector is defined to attached a timeline to a case outside of the case page.
+  // FYI, if you are in the case page we only use handleOnTimelineChange to attach a timeline to a case.
+  useEffect(() => {
+    const currentValue = form.getFormData()[fieldName];
```

Comment:
@sebelga could you take a look at that?

Reply:
It would be better to use the `useFormData` hook and directly provide the field value to this hook. For example, the `AddComment` would be

```js
const { form } = useForm<CommentRequest>({
  defaultValue: initialCommentValue,
  options: { stripEmptyFields: false },
  schema,
});
const { setFieldValue, reset, submit } = form;

// Get the updated field value
const [{ comment }] = useFormData({ form, watch: ['comment'] });

// Pass it to the hook
const { handleCursorChange, handleOnTimelineChange } = useInsertTimeline<CommentRequest>(comment);

// I am not sure I understand what this does...
// If you need to format the field value on each keystroke, use a "formatter" on the field
// If you need to format the field when initiated, use a "deserializer" on the field.
const addQuote = useCallback(
  (quote) => {
    setFieldValue('comment', `${comment}${comment.length > 0 ? '\n\n' : ''}${quote}`);
  },
  [comment, setFieldValue]
);
```

Reply:
@sebelga Thank you for your feedback! The `addQuote` function takes the content of a comment and make it a quote. This will only happen when the user clicks the add quote button and not on each keystroke or at the initiation of the field.

Reply:
I see, thanks for the explanation @cnasikas